### ISSUE TITLE
git ship: commit with title of pull request

### ISSUE THREAD

**Issue by kevgo** [git-town/git-town#1537]
As a code reviewer approving a feature branch
I want that the changes are shipped exactly as I reviewed them
So that I don't have to re-review the shipped changes again.

- with shipping with a CodeHostingDriver, `git ship` should ship with the title of the pull request and not allow the user to edit the commit message of the squashed commit

**Comment by blaggacao** [comment]
I think morenoften than not obly the reviewed has right to ship (push onto a protected branch) therefore, I think it would be better to load PR title and body into the commit message template.

This would act as kind of a prepare commit message hook and present the suggestion to the shipper.

In such variant, it would combine best of both worlds and allow for more versatile workflows.

**Comment by kevgo** [comment]
I think the default should be that developers ship their own branches. The reason is DevOps: developers should be responsible for the code they write through the entire lifecycle of it. If their code has a merge conflict, they (not their reviewers) should resolve it. If they break the build after merging, they (not their reviewers) should fix it. They (not their reviewers) should publish code to production. They (not their reviewers) should check whether their code causes performance or stability regressions in production. Reviewers are senior engineers, we don't want to make them bottlenecks. They should their time as efficiently as possible, which is to just review code, not sync, ship, and launch it for their teammates. Those are activities that that less experienced devs should do so that they learn the ropes.

The only exception to this rule are contributions from external developers that have no rights to merge pull requests, which we often find in open-source. In those case, a maintainer has to merge. In those situations, the merge will happen via a web UI, like the Gitea or GitHub UI.

**Comment by blaggacao** [comment]
Good point! You are teaching me something, here. It's a great pleasure to be contributing to this project. I almost exclusively worked on oss projects in the past.

Still, I feel enforcing is a taint too harsh for some use cases or practices, but putting it as the most sensible default in the commit message would be just about perfect.
